src/t_hash.c

```diff
@@ -1034,10 +1044,18 @@ SetExRes hashTypeSetExpiryHT(HashTypeSetEx *exInfo, sds field, uint64_t expireAt
     dictSetKey(ht, existingEntry, hfNew);
 
 
-    /* if expiration time is in the past */
-    if (unlikely(checkAlreadyExpired(expireAt))) {
-        hashTypeDelete(exInfo->hashObj, field, 1);
-        exInfo->fieldDeleted++;
+    /* If expired, then delete the field and propagate the deletion.
+     * If replica, continue like the field is valid */
+    if (unlikely((checkAlreadyExpired(expireAt)) && (server.masterhost == NULL))) {
```

Comment:
Please see implementation of `checkAlreadyExpired()` function. It already contains `server.masterhost` check in it..

Reply:
i refactored this code. pls review again.